fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -295,6 +303,139 @@ public CommandProcessingResult disburseLoan(Long loanId, JsonCommand command, Bo
         return disburseLoan(loanId, command, isAccountTransfer, false);
     }
 
+    @Transactional
+    @Override
+    public CommandProcessingResult disburseLoanToLinkedAccount(Long loanId, JsonCommand command, Boolean isAccountTransfer) {
+        Boolean isWithoutAutoPayment = false;
```

Comment:
Wow this is super complicated. I understand the existing methods are already there in  this complicated form but let's not introduce a new complex thing here.

Let's follow the clean code principles, have separate methods for the individual logical pieces. Have a completely separate class that handles the entire process, let's have small methods, etc etc.